docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -511,6 +511,88 @@ quarkus.http.auth.permission.bearer.auth-mechanism=bearer
 
 Ensure that the value of the `auth-mechanism` property matches the authentication scheme supported by `HttpAuthenticationMechanism`, for example, `basic`, `bearer`, or `form`.
 
+=== Endpoint-specific authentication mechanisms
+
+It is possible to select authentication mechanism specific for each Jakarta REST endpoint and Jakarta REST resource with annotation.
+This feature is only enabled when <<proactive-auth>> is disabled, for authentication selection must happen after a REST endpoint has been matched.
+Here is how you can select <<openid-connect-authentication>> mechanism per a REST endpoint basis:
+
+[source,properties]
+----
+quarkus.http.auth.proactive=false
+
+# combine HTTP permissions with the endpoint-specific authentication mechanism
```

Comment:
IMHO we should not start with giving an HTTP security example when introducing this feature, I'd add a subsection like `==== How to combine it with HTTP security policy` and give this `applies-to` example there.
In the introduction itself, I propose to give `BasicAuthentication` and `AuthorizationCodeFlow` as users did ask for this kind of set up before

Reply:
Sure, combination with the HTTP Security Policies is rather for compatibility than recommended use case. I'll move it.